### ISSUE TITLE
This project is dead

### ISSUE THREAD

**Issue by Tmp341** [riderkick/FMD#1713]
I've written something like this in #819 Then after some time, @kmvi came by and started fixing some issues. But he is gone now too. This project is dead now.

@riderkick @kavin-90 and @kmvi, you guys are all gone. If you guys are reading this, please give me a position, so i can write some last words and close this project.

To all users, until new developers show up, like i said, this project is dead. Please check other downloaders.

**Comment by megacin** [comment]
YES al most web site change tchnique of they website

**Comment by Twoearle** [comment]
FMD is open source, anyone can contribute to the project or fork their own version. That being said, you need the skills to do so and that might be the biggest issue.

**Comment by anfgomezmo** [comment]
The project is not that dead, the lua modules are working and we can edit them if needed. The main issue right now is trying to fix KissManga and MangaFox because those are embedded in the main core so you will need to recompile the whole program. We just need organization, I will start checking the main code will take me some days. Also, I suggest that we edit the lua modules if there are some pages down. 

Cloudflare could be problematic because they did a major security update in Febraury but solutions will appear in the next weeks, I hope. We need to have patience. 

The other problem, we need someone that can validate the Pull requests as soon as possible.

**Comment by Tmp341** [comment]
> The project is not that dead, 

I was talking about it's developing side. All the developers left it without a word. 😃 

We need an active approver on pull requests.

**Comment by ChocolateOtaku** [comment]
> ... the LUA modules are working and we can edit them if needed ...

Yeah, but to bad the LUA scripts are updated from a central server that is this dead github repository

Have fun with the emerging LUA hell where everyone is trying to fix the LUA scripts and sharing their various modified LUA scripts... as a user i'm not willing to crawl the tickets to find working LUA files

And not even mentioning the annoying cookie copy paste issues that i saw in the recent tickets... sorry but this now seems to be hackware and no longer software, glad im migrated quite a while ago :D

**Comment by anfgomezmo** [comment]
Emerging the LUA hell is easy; we need to change and update a few code lines in the program and recompile it. The real problem is defining if continue with this repo or forking a new one and start working in it.  

About the cookie issue, please understand that the webpages change their structure a lot of times to block this kind of programs. Even the paid ones are having a similar problem.

**Comment by ChocolateOtaku** [comment]
This will work as long as the next developer(s) abandon this project and then the cycle will start over...

I never had to touch any Cookies or Keys with HakuNeko and its still working fine (and its not a paid software either)

**Comment by ghost** [comment]
I can only repair parts of LUA file, so if there is a problem when you download the comic, please let me know, I might be able to fix it a little

Sorry my English language bad

**Comment by anfgomezmo** [comment]
HakuNeko uses JS and it is a plus. Also, it is cloud-hosted, that's why you do not need to change cookies or else because someone is updating the files (must be Ronny). If he gets tired one day, the same could happen. I suggest all the users to use the program they feel comfortable.

**Comment by Tmp341** [comment]
> If he gets tired one day, the same could happen.

It will happen.

**Comment by SDXC** [comment]
I have a suggestion for now:
As I am currently looking into the code and trying to understand the mechanics and everything there is a possibility that I'll take over the project.

For advanced changes it will take a bit of time, but I can start with a fork of the project and take over all the pull requests and available custom lua fixes and make a first release. Of course I'll also update the online checks so they point to the new repository. New issues will be handled in the new repo and pull requests will be accepted more frequently. So even if I am unable to bring fixes fast, if someone makes a pull request that works, I'll merge it.

Please let me know what you think (voting by thumbs up is also okay).

**Comment by SDXC** [comment]
Please follow this project now:
https://github.com/fmd-project-team/FMD

New issues should also be opened there.

I already merged and did some fixes myself.
A first version of my forked FMD will probably be released this weekend.

There're some more fixes I want to implement and I have to make a forced update to the config for the lua download URLs.

**Comment by megacin** [comment]
Slm khoya momkin mosa3ada

**Comment by anfgomezmo** [comment]
I am fine with the merging, I was checking the code too. I think that we need to take out kissmanga and mangafox from the core and create a lua

**Comment by SDXC** [comment]
"Lualizing" the old modules will be one of my main tasks in the next few days, I think.

**Comment by 27136773** [comment]
> > If he gets tired one day, the same could happen.
> 
> It will happen.

No developer will stick around forever (lifespan, private reasons, ...).
For a development project to survive, it is important to attract new developers.
There are various ways to make the project attractive to new developers, here are some tips, that from my personal point of view, may help:

- Provide a well written developer documentation
- Apply [clean code](https://www.investigatii.md/uploads/resurse/Clean_Code.pdf) principles (no developer wants to work with code that is difficult to understand)
- Usa a programming language/framework and development tools that are [popular](https://www.benfrederickson.com/ranking-programming-languages-by-github-users/) (no developer wants to struggle with a new language that nobody else uses, especially if it is not in need for companies in case of job search)
- Promote the product (no developer wants to work on a project that has only 12 users)

That said, HakuNeko has also trouble to attract new developers, even with using popular JavaScript and applying clean code principles...

Good luck with the future development

**Comment by Tmp341** [comment]
> No developer will stick around forever (lifespan, private reasons, ...).
> For a development project to survive, it is important to attract new developers.

Yeah. Some people actually trying to do something, but there is no active owner, who can approve PRs in here. That's FMD's problem, not finding new developers. I don't know anything about coding, and i don't want to start it with Pascal. There are C, JavaScript, Python etc could be useful for me.

**Comment by SDXC** [comment]
It's now official:
[Free Manga Downloader - 0.9.160.0](https://github.com/fmd-project-team/FMD/releases/tag/0.9.160.0)

**Comment by kavin-90** [comment]
well, i'm newbie in pascal so can't do much.

**Comment by SDXC** [comment]
@kavin-90 When I made that fork my Pascal skills were exactly 0% and I can absolutely understand why nobody wants to learn this language^^

Maybe porting FMD to a new platform might be a good thing but takes a lot of time.

**Comment by kavin-90** [comment]
I know php only so dunno. i just copy/paste pascal stuff to add Web Young Jump, Comico etc
